### DISCUSSION TITLE
NJU 编译原理 - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#583] [Announcements]
# NJU 编译原理 - CS自学指南

CS自学指南

https://csdiy.wiki/%E7%BC%96%E8%AF%91%E5%8E%9F%E7%90%86/NJU-Compilers/

**Comment by KindofCrazy** [comment]
蚂蚁蚂蚁！

**Reply by wuc9521** [reply]
蚂蚁蚂蚁！

**Reply by Luke-Evan** [reply]
蚂蚁蚂蚁!

**Reply by 4ever-xxxl** [reply]
蚂蚁蚂蚁！

**Reply by Aron00123** [reply]
蚂蚁蚂蚁！

**Reply by Himalian** [reply]
蚂蚁蚂蚁！

**Reply by ddddddavid66** [reply]
蚂蚁蚂蚁！

**Comment by Qi-Zhan** [comment]
南大还有一个编译原理课程（个人认为更好难度更大）

https://cs.nju.edu.cn/changxu/2_compiler/index.html

**Reply by TekkenSteve** [reply]
这个貌似是拔尖计划的？看看谭添老师（也就是教软件分析也叫静态分析的其中一位老师）的南京大学主页，介绍那里就有2024年的编译原理授课计划，可能中途有什么教学变更了，班级号是一样的（后缀有写是理科强化班），但突然谭老师的教学动态就中断了。如果也学过太阿的，也可以看看谭老师往年的教学，这样衔接更流畅。

**Reply by TekkenSteve** [reply]
所以应该只是教学受众不一样，难度有差异很正常，而且这个课程老师并没有给到校外学生也那么充足的教学资源，要选这个的慎重。

**Reply by Jackcuii** [reply]
许畅老师一直以来负责这个实验，从今年开始tt老师也开始负责拔尖班的这门课授课。tt老师的软件分析是开源的，欢迎来学习!

**Reply by Lhcfl** [reply]
但是这门课的实验就比较抽象，这门课用的语言是C，给编码带来了毫无必要的难度增加，尤其对于习惯了现代编程语言的同学更是一种折磨

**Reply by SweetGargamel** [reply]
我是cs大班的（现大二），想问一下编译原理这个课下学期如果是许老师上的话有没有比较配套的网课可以看的？软院蚂蚁老师课如果用来应付期末怎么样?以及lab的压力怎么样。

因为大班这个课是在一个9选3的课程组在考虑要不要选（以后不打算走PL，也不打算走科研就是想本科找后端开发，提前选课纯粹是为了凑学分这样大三能空出来一些时间，想找事情不那么多的）

**Reply by Qi-Zhan** [reply]
我也没上过哈哈, lab的压力应该是挺大的, 不过现在有AI, 说实话想水都可以水吧.

**Comment by sarendis56** [comment]
十分的支持并且推荐

**Comment by chaos-warzh** [comment]
十分推荐, 这门课的实验内容很丰富, 蚂蚁老师的讲授也十分细致

**Comment by kyiree** [comment]
希望大家能够成为一名真正的超级工程师

**Comment by GoodNightmj** [comment]
有南京大学其他课程的教学资源分享吗
没有找到q w q

**Reply by Qi-Zhan** [reply]
https://cs.nju.edu.cn/changxu/2_compiler/index.html

**Comment by Weakstupidchatbox** [comment]
有个问题想请教一下大家，我们学校开了信号处理的课，我个人对这节课没有什么兴趣但是可以拿学分，所以想请问一下学习这方面的知识对以后有什么好处，以后可以从事什么样的工作或研究，还有就是如果想要自学的话有没有比较好的英文或者中文的课程，麻烦各位大佬了

**Reply by kyiree** [reply]
如果将来做通信相关的工作或者研究，这门课是必修课例如前几年很火的5g

**Comment by nooblesone** [comment]
为了学这网站的课现在正在狂补英语

**Comment by sleeper0122** [comment]
今天照着手册下载实验代码的时候，提示502 BadGateway，oj网站也访问不了，是在维护还是说关了呀😢

**Reply by sleeper0122** [reply]
现在好了嘿嘿😍

**Comment by EdogawaAi** [comment]


**Reply by Akane0238** [reply]
插个眼，期待上交的编译原理😚！

**Comment by WUADN** [comment]
对于过了截止日期的，老师说应该创建分支为labN-copy，同时submit.sh的$ID也应该同步更改.如下代码，$ID添加-copy后缀
# Construct assignmentId
ANTLRTMP="labN"
copy="-copy"
BRANCH=$(git symbolic-ref --short -q HEAD)
NUMBER=$(echo $BRANCH | tr -cd "[0-9]")
ID="$(echo $ANTLRTMP | sed "s/N/${NUMBER}/g")$copy"

**Reply by zhaoxinyu-xuxiaoguai** [reply]
过了截止日期还需要申请账号吗？申请好多天了还是没回应
